sphinx/builders/linkcheck.py

```diff
@@ -636,12 +640,29 @@ def run(self, **kwargs: Any) -> None:
         for imgnode in self.document.traverse(nodes.image):
             uri = imgnode['candidates'].get('?')
             if uri and '://' in uri:
+                newuri = self.app.emit_firstresult('linkcheck-process-uri', uri)
+                if newuri:
+                    uri = newuri
+
```

Comment:
Not sure if it’s worth checking anchors on image nodes?

Reply:
No. I added an event for common usage and a custom event handler for GitHub anchors in this PR. The "linkcheck-process-uri" event is the former one. It's not only for anchors. Extensions can modify the URIs as they like (as `rewrite_github_anchor()` does). So the event should be emitted for images too.